crates/nix_rs/src/version_spec.rs

```diff
@@ -0,0 +1,204 @@
+//! Rust module defining the spec for [NixVersion]
+
+use std::{fmt, str::FromStr};
+
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+use crate::version::{BadNixVersion, NixVersion};
+
+/// Specifies a version requirement for Nix
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub enum NixVersionSpec {
+    /// Version must be greater than the specified version
+    Gt(NixVersion),
+    /// Version must be greater than or equal to the specified version
+    Gteq(NixVersion),
+    /// Version must be less than the specified version
+    Lt(NixVersion),
+    /// Version must be less than or equal to the specified version
+    Lteq(NixVersion),
+    /// Version must not equal the specified version
+    Neq(NixVersion),
+}
+
+/// A collection of version requirements
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct NixVersionReq {
+    /// List of version specifications
+    pub specs: Vec<NixVersionSpec>,
+}
+
+/// Errors that can occur while parsing or validating version specifications
+#[derive(Error, Debug)]
+pub enum BadNixVersionSpec {
+    /// Error occurred while parsing the Nix version
+    #[error("Nix version error while parsing spec: {0}")]
+    NixVersion(#[from] BadNixVersion),
+
+    /// The version requirements cannot be satisfied together
+    #[error("Version bounds are not satisfiable")]
+    UnsatisfiableBounds,
+
+    /// An unknown comparison operator was used
+    #[error("Unknown operator in the Nix version spec: {0}")]
+    UnknownOperator(String),
+}
+
+impl NixVersionReq {
+    /// Parses a comma-separated string of version requirements
+    ///
+    /// # Example
+    /// ```
+    /// use nix_rs::version_spec::NixVersionReq;
+    /// let req = NixVersionReq::parse(">2.8, <3.0, !=2.9").unwrap();
+    /// ```
+    pub fn parse(s: &str) -> Result<Self, BadNixVersionSpec> {
+        let specs = s
+            .split(',')
+            .map(str::trim)
+            .map(NixVersionSpec::from_str)
+            .collect::<Result<Vec<_>, _>>()?;
+
+        Self::check(&specs)?;
+        Ok(NixVersionReq { specs })
+    }
+
+    /// Checks if all version specifications are satisfiable
+    ///
+    /// Returns Err if the requirements are logically impossible to satisfy
+    fn check(specs: &[NixVersionSpec]) -> Result<(), BadNixVersionSpec> {
```

Comment:
I was confused by this function for a moment.

```suggestion
    fn ensure_logically_sound(specs: &[NixVersionSpec]) -> Result<(), BadNixVersionSpec> {
```

Reply:
I have now removed this function, as per our huddle discussion, to keep the interface simple.